js/ad_manager_vast.js

```diff
@@ -697,24 +739,29 @@ OO.Ads.manager(function(_, $) {
      * If a non-linear Ad is found then it is parsed and added to the timeline via the addToTimeline function.
      * @public
      * @method Vast#_handleNonLinearAd
+     * @param {object} ad The ad object
      * @param {object} adLoaded The ad that was loaded
+     * @param {object} params Additional params
      * @returns {boolean} True if the load was successful and a stream was found otherwise false.
      */
-    this._handleNonLinearAd = function(adLoaded) {
+    this._handleNonLinearAd = function(ad, adLoaded, params) {
       // filter our playable stream:
-      var firstNonLinearAd = _.find(this.inlineAd.ads, function(v){ return !_.isEmpty(v.nonLinear.url); }, this);
-      if (!firstNonLinearAd) { return false; }
-      var adURL = firstNonLinearAd.nonLinear.url;
-      this.vastAdUnit.streamUrl = adURL;
-      _.extend(this.vastAdUnit.data, firstNonLinearAd);
-      this.vastAdUnit.data.tracking = firstNonLinearAd.nonLinear.tracking;
-
-      if (this.vastAdUnit.streamUrl == null) {
+      if (_.isEmpty(ad.nonLinear.url)) { return false; }
+      params = params ? params : {};
+      var adURL = ad.nonLinear.url;
+      var vastAdUnit = { data: {}, vastUrl: this.vastUrl, maxBitrateStream: null };
```

Comment:
Nitpicky, but I think it's slightly more readable to declare the object literal like:
```
var vastAdUnit = {
  data: {},
  vastUrl: this.vastUrl,
  maxBitrateStream: null
};
```